modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,32 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
+				var webp_sizes = window.wpPerfLabWebpSizes;
+				var size_keys = Object.keys( webp_sizes );
```

Comment:
See my other comment. We should iterate through the `sizes` from the REST API response and check for each one whether both a `image/webp` and `image/jpeg` version available, and if so we can do the replacement.

We don't need to know which sizes _should_ have a WebP version available here.

Reply:
Thank you for the review. The PR was updated with the logic.